src/librustc_typeck/diagnostics.rs

```diff
@@ -1489,22 +1489,42 @@ For information on the design of the orphan rules, see [RFC 1023].
 "##,
 
 E0118: r##"
-Rust can't find a base type for an implementation you are providing, or the type
-cannot have an implementation. For example, only a named type or a trait can
-have an implementation:
+You're trying to implement something which isn't a struct nor an enum.
```

Comment:
This sentence doesn't make sense.

"You're trying to write an inherent implementation for something which isn't a struct or an enum"